### PR TITLE
Improved logging and connection timeout field added.

### PR DESCRIPTION
**Pull request by vikasrathee-cs**
Improved logging and connection timeout field added.

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java [hunk 1]
```diff
@@ -51,14 +55,46 @@ public static PartnerConnection getPartnerConnection(AuthenticatorCredentials cr
...
+   */
+  public static String getSalesforceErrorMessageFromException(Exception e) {
+    if (e instanceof IApiFault) {
+      return ((IApiFault) e).getExceptionMessage();
+    } else {
+      return e.getMessage();
+    }
+  }
+
+  public static OAuthInfo getOAuthInfo(BaseSalesforceConfig config, FailureCollector collector) {
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java:93]: Add java doc

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormatProvider.java [hunk 2]
```diff
@@ -50,8 +50,8 @@ public SalesforceOutputFormatProvider(SalesforceSinkConfig config) {
       .put(SalesforceSinkConstants.CONFIG_OPERATION, config.getOperation())
       .put(SalesforceSinkConstants.CONFIG_ERROR_HANDLING, config.getErrorHandling().getValue())
       .put(SalesforceSinkConstants.CONFIG_MAX_BYTES_PER_BATCH, config.getMaxBytesPerBatch().toString())
-      .put(SalesforceSinkConstants.CONFIG_MAX_RECORDS_PER_BATCH, config.getMaxRecordsPerBatch().toString());
-
+      .put(SalesforceSinkConstants.CONFIG_MAX_RECORDS_PER_BATCH, config.getMaxRecordsPerBatch().toString())
+      .put(SalesforceConstants.CONFIG_CONNECT_TIMEOUT, config.getConnectTimeout().toString());      ;
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormatProvider.java]: nit: remove additional `;`

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java [hunk 3]
```diff
@@ -46,7 +44,9 @@ public RecordWriter<NullWritable, CSVRecord> getRecordWriter(TaskAttemptContext
     try {
       return new SalesforceRecordWriter(taskAttemptContext);
     } catch (AsyncApiException e) {
-      throw new RuntimeException("There was issue communicating with Salesforce", e);
+      throw new RuntimeException(
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java:47]: Is the cause captured here correctly? Do we need to check for IApiFault ?

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java [hunk 4]
```diff
@@ -79,7 +79,9 @@ public void commitJob(JobContext jobContext) {
           String jobId = conf.get(SalesforceSinkConstants.CONFIG_JOB_ID);
           SalesforceBulkUtil.closeJob(bulkConnection, jobId);
         } catch (AsyncApiException e) {
-          throw new RuntimeException("There was issue communicating with Salesforce", e);
+          throw new RuntimeException(
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java:82]: Is the cause captured here correctly? Do we need to check for IApiFault ?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java [hunk 5]
```diff
@@ -184,7 +185,44 @@ public void validate(FailureCollector collector) {
...
+        }
+      } catch (InvalidConfigException e) {
+        collector.addFailure(e.getMessage(), null).withConfigProperty(e.getProperty());
+      }
+    }
+    validateSchema(collector);
+    validatePKChunk(collector, oAuthInfo);
+  }
+
+  public void validateSOQLQueryParameters(FailureCollector collector) {
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java:210]: Add doc

File: src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceETLTest.java [hunk 6]
```diff
@@ -70,7 +70,8 @@ public abstract class BaseSalesforceETLTest extends HydratorTestBase {
                                                              "https://login.salesforce.com/services/oauth2/token");
   protected static final String SECURITY_TOKEN = System.getProperty("salesforce.test.securityToken",
                                                                "");
-
+  protected static final String CONNECT_TIMEOUT = System.getProperty("salesforce.test.connectTimeout",
+                                                                  "30000");
```
Comment (greeshmaswaminathan) [inline src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceETLTest.java]: Incorrect formatting here.

File: src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchSinkETLTest.java [hunk 7]
```diff
@@ -38,10 +41,10 @@ public class SalesforceBatchSinkETLTest extends BaseSalesforceBatchSinkETLTest {
   public void testInsertAccount() throws Exception {
     String sObject = "Account";
     Schema schema = Schema.recordOf("output",
-                                    Schema.Field.of("Name", Schema.of(Schema.Type.STRING)),
-                                    Schema.Field.of("NumberOfEmployees", Schema.of(Schema.Type.INT)),
-                                    Schema.Field.of("ShippingLatitude", Schema.of(Schema.Type.DOUBLE)),
-                                    Schema.Field.of("ShippingLongitude", Schema.of(Schema.Type.DOUBLE))
+      Schema.Field.of("Name", Schema.of(Schema.Type.STRING)),
```
Comment (greeshmaswaminathan) [inline src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchSinkETLTest.java]: Incorrect formatting, check other files too.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java (hunk 3 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java:47]: IAPIFault will work for ConnectionException and  AsyncApiException return proper error message.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java (hunk 4 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormat.java:82]: IAPIFault will work for ConnectionException and  AsyncApiException return proper error message.

File: src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java (hunk 1 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java:93]: Added.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormatProvider.java (hunk 2 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceOutputFormatProvider.java]: removed

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java (hunk 5 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java:210]: done

File: src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceETLTest.java (hunk 6 above)
Comment (vikasrathee-cs) [reply src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceETLTest.java]: changed.

File: src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchSinkETLTest.java (hunk 7 above)
Comment (vikasrathee-cs) [reply src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchSinkETLTest.java]: changed

**Review by greeshmaswaminathan — APPROVED** [review]

**Pull request merged**